bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,384 @@
+            sendMessage(YioRemoteMessages.AUTHENTICATE_MESSAGE, localConfig.accessToken);
+            yioremotedockactualstatus = YioRemoteDockHandleStatus.AUTHENTICATION_PROCESS;
+        } else if (yioremotedockactualstatus.equals(YioRemoteDockHandleStatus.AUTHENTICATION_PROCESS)) {
+            if (getbooleanauthenticationok()) {
+                yioremotedockactualstatus = YioRemoteDockHandleStatus.AUTHENTICATION_COMPLETE;
+                updateStatus(ThingStatus.ONLINE);
+                startwebsocketpollingthread();
+                if (authenticationjob != null) {
+                    authenticationjob.cancel(true);
+                }
+            } else {
+                yioremotedockactualstatus = YioRemoteDockHandleStatus.AUTHENTICATION_FAILED;
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Connection lost no ping from YIO DOCK");
+            updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+        }
+    }
+
+    private void startwebsocketpollingthread() {
+        Runnable websocketpollingthread = new Runnable() {
+            @Override
+            public void run() {
+                if (yioremotedockactualstatus.equals(YioRemoteDockHandleStatus.AUTHENTICATION_COMPLETE)) {
+                    if (getbooleanheartbeat()) {
+                        updateChannelString(GROUP_OUTPUT, STATUS_STRING_CHANNEL,
+                                ircodereceivedhandler.getcode() + ircodereceivedhandler.getformat());
+                        logger.debug("heartbeat ok");
+                        sendMessage(YioRemoteMessages.HEARTBEAT_MESSAGE, "");
+                    } else {
+                        yioremotedockactualstatus = YioRemoteDockHandleStatus.CONNECTION_FAILED;
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "Connection lost no ping from YIO DOCK");
+                        updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+                        if (websocketpollingjob != null) {
+                            websocketpollingjob.cancel(true);
+                        }
+                    }
+                } else {
+                    if (websocketpollingjob != null) {
+                        websocketpollingjob.cancel(true);
+                    }
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Connection lost no ping from YIO DOCK");
+                    updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+                }
+            }
+        };
+        websocketpollingjob = scheduler.scheduleWithFixedDelay(websocketpollingthread, 0, 30, TimeUnit.SECONDS);
+    }
+
+    public boolean getbooleanauthenticationok() {
+        return authenticationok;
+    }
+
+    public boolean getbooleanheartbeat() {
```

Comment:
You could clarify in the method's name, that this isn't simply a getter.
```suggestion
    public boolean getAndResetHeartbeat() {
```

Reply:
dones